src/TypeFormatter/LiteralUnionTypeFormatter.ts

```diff
@@ -72,13 +73,29 @@ export class LiteralUnionTypeFormatter implements SubTypeFormatter {
 export function isLiteralUnion(type: UnionType): boolean {
     return type
         .getFlattenedTypes()
-        .every((item) => item instanceof LiteralType || item instanceof NullType || item instanceof StringType);
+        .every(
+            (item) =>
+                item instanceof LiteralType ||
+                item instanceof NullType ||
+                item instanceof StringType ||
+                item instanceof EnumType,
+        );
 }
 
-function getLiteralValue(value: LiteralType | NullType): LiteralValue | null {
-    return value instanceof LiteralType ? value.getValue() : null;
+function getLiteralValues(value: LiteralType | EnumType | NullType): readonly (LiteralValue | null)[] {
+    if (value instanceof EnumType) {
+        return value.getValues();
+    } else if (value instanceof LiteralType) {
+        return [value.getValue()];
+    }
+    return [null];
```

Comment:
Why not just return `[]` and change the type to only allow `readonly LiteralValue[]`?